R/teal_modifiers.R

```diff
@@ -0,0 +1,193 @@
+#'   data = teal_data(IRIS = iris, MTCARS = mtcars),
+#'   modules = modules(example_module())
+#' ) |>
+#'   modify_title(title = "Custom title")
+#'
+#' if (interactive()) {
+#'   shinyApp(app$ui, app$server)
+#' }
+modify_title <- function(
+    x,
+    title = "teal app",
+    favicon = "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/nest.png") {
+  checkmate::assert_multi_class(x, "teal_app")
+  checkmate::assert_multi_class(title, c("shiny.tag", "shiny.tag.list", "html", "character"))
+  checkmate::assert_string(favicon)
+  teal_replace_ui(
+    x,
+    "#teal-app-title",
+    tags$head(
+      tags$title(title),
+      tags$link(
+        rel = "icon",
+        href = favicon,
+        sizes = "any"
+      )
+    )
+  )
+}
+
+#' Add a Header to `teal` Application
+#'
+#' @rdname teal_modifiers
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris),
+#'   modules = modules(example_module())
+#' ) |>
+#'   modify_header(element = tags$div(h3("Custom header")))
+#'
+#' if (interactive()) {
+#'   shinyApp(app$ui, app$server)
+#' }
+modify_header <- function(x, element = tags$p()) {
+  checkmate::assert_multi_class(element, c("shiny.tag", "shiny.tag.list", "html", "character"))
+  teal_replace_ui(x, "#teal-header-content", element)
+}
+
+#' Add a Footer to `teal` Application
+#'
+#' @rdname teal_modifiers
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris),
+#'   modules = modules(example_module())
+#' ) |>
+#'   modify_footer(element = "Custom footer")
+#'
+#' if (interactive()) {
+#'   shinyApp(app$ui, app$server)
+#' }
+modify_footer <- function(x, element = tags$p()) {
+  checkmate::assert_multi_class(element, c("shiny.tag", "shiny.tag.list", "html", "character"))
+  teal_replace_ui(x, "#teal-footer-content", element)
+}
+
+#' Add a Landing Popup to `teal` Application
+#'
+#' @description Adds a landing popup to the `teal` app. This popup will be shown when the app starts.
+#' The dialog blocks access to the application and must be closed with a button before the application can be viewed.
+#'
+#' @param x (`teal_app`) A `teal_app` object created using the `init` function.
+#' @param title (`character(1)`) Text to be displayed as popup title.
+#' @param content (`character(1)`, `shiny.tag` or `shiny.tag.list`) with the content of the popup.
+#'  Passed to `...` of `shiny::modalDialog`.
+#' @param buttons (`shiny.tag` or `shiny.tag.list`) Typically a `modalButton` or `actionButton`.
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris, MTCARS = mtcars),
+#'   modules = modules(example_module())
+#' ) |>
+#'   add_landing_popup(
+#'     title = "Welcome",
+#'     content = "This is a landing popup.",
+#'     buttons = modalButton("Accept")
+#'   )
+#'
+#' if (interactive()) {
+#'   shinyApp(app$ui, app$server)
+#' }
+add_landing_popup <- function(
+    x,
+    title = NULL,
+    content = NULL,
+    buttons = modalButton("Accept")) {
```

Comment:
I think function could be named `add_landing_modal`. Arguments are passed to `modalDialog` and I think they should be inherited. `modalDialog` has also `size`, `easyClose`, `fade` arguments - these we can consider to add in the future (explicitly or through ...). 

```suggestion
    title = NULL,
    content = NULL,
    footer = modalButton("Accept")) {
```

Reply:
I think there is no harm in adding `...` even now. I'll add it.